extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -602,41 +606,45 @@ public static void deleteTable(File path) {
      * key} order) location found will be used to infer schema.
      *
      * <p>
-     * Delegates to one of {@link #readSingleFileTable(File, ParquetInstructions)},
+     * Delegates to one of {@link #readSingleFileTable(URI, ParquetInstructions)},
      * {@link #readPartitionedTableWithMetadata(File, ParquetInstructions)},
      * {@link #readFlatPartitionedTable(File, ParquetInstructions)}, or
      * {@link #readKeyValuePartitionedTable(File, ParquetInstructions)}.
      *
-     * @param source The source file or directory
+     * @param source The URI for source file or directory
      * @param instructions Instructions for reading
      * @return A {@link Table}
      */
     private static Table readTableInternal(
-            @NotNull final File source,
+            @NotNull final URI source,
             @NotNull final ParquetInstructions instructions) {
-        final Path sourcePath = source.toPath();
+        if (S3_URI_SCHEME.equals(source.getScheme())) {
+            return readSingleFileTable(source, instructions);
+        }
+        final Path sourcePath = Path.of(source.getRawPath());
```

Comment:
I don't understand why we are using raw path, you can create Path from URI: `java.nio.file.Path#of(java.net.URI)` (assuming there is a proper Filesystem).